### PR TITLE
Supports to build PAL tests under x86/Linux

### PR DESCRIPTION
**Pull request by parjong**

### CODE REVIEW

**Comment by parjong** [comment]
CC @seanshpark

**Comment by parjong** [comment]
@dotnet-bot test Windows_NT x64 Debug Build and Test please

File: src/pal/tests/CMakeLists.txt [hunk 1]
```diff
@@ -23,7 +27,7 @@ elseif(PAL_CMAKE_PLATFORM_ARCH_ARM64)
   add_definitions(-DBIT64=1)
   add_definitions(-D_WIN64=1)
 else()
-  message(FATAL_ERROR "Only ARM, ARM64 and AMD64 is supported")
+  message(FATAL_ERROR "Only ARM, ARM64, I386, and AMD64 is supported")
```
Comment (parjong) [inline src/pal/tests/CMakeLists.txt]: I'm also confused. build.sh uses x86 and x64 but CMakeLists.txt usually use I386 and AMD64, respectively. 

Here I followed the usual convention in CMakeLists.txt.

**Comment by RussKeldorph** [comment]
Please ignore the x86 compatjit and x86 legacy_backend jobs if they fail.  They have been removed from future PRs.

File: src/pal/tests/CMakeLists.txt (hunk 1 above)
Comment (janvorli) [inline src/pal/tests/CMakeLists.txt]: We already use CLR_CMAKE_TARGET_ARCH_I386 in the CMakeLists.txt, so this matches the change.

**Pull request merged**